### PR TITLE
Align embedded and XML defaults for HttpConfiguration

### LINKED ISSUE #11947: Jetty12: HttpConfiguration#_relativeRedirectAllowed flipped to true per default
**Jetty Version**

jetty-12

**Jetty Environment**

ee10

**Java Version**

Java 21

**Question**

During migration from Jetty 11 to Jetty 12 (ee10) we started to see some test failures.

Particularly, the `Location` header wasn't set to absolute URL after the redirect.
Particularly, the if statement wasn't `true` any more after migration to Jetty 12:

jetty-server/src/main/java/org/eclipse/jetty/server/Response.java
```java
  // if relative redirects are not allowed?
  if (!request.getConnectionMetaData().getHttpConfiguration().isRelativeRedirectAllowed())
  {
    // make the location an absolute URI
    location = URIUtil.newURI(uri.getScheme(), Request.getServerName(request), Request.getServerPort(request), location, null);
  }
```

We've investigated the reason for this test failure and concluded, that there was an undocumented behavioural change in Jetty 12 in `HttpConfiguration` class in context of this issue #11014 and PR #11019.

To fix the behaviour in our case we have to unflip this option, by setting (see [1] for the whole change):

```java
  private HttpConfiguration defaultConfig(int requestHeaderSize) {
    HttpConfiguration config = new HttpConfiguration();
    config.setRequestHeaderSize(requestHeaderSize);
    config.setRelativeRedirectAllowed(false);
    config.setSendServerVersion(false);
    config.setSendDateHeader(true);
    return config;
  }
```

Question 1:
Can this behavioural change be better documented?

Question 2:
Why the default `jetty.xml` still has `default="false"`?

jetty-server/src/main/config/etc/jetty.xml
```xml
[...]
<Set name="relativeRedirectAllowed"><Property name="jetty.httpConfig.relativeRedirectAllowed" default="false"/></Set>
[...]
```

[1] https://gerrit-review.googlesource.com/c/gerrit/+/424580

### PR DESCRIPTION
**Pull request by gregw**
Fix #11947 by aligning embedded and XML HttpConfiguration defaults

### CODE REVIEW

**Review by lachlan-roberts — DISMISSED** [review]
I think the test failures are not a flake, the `SiweTest` was expecting the full URI in the redirect but now only gets the relative redirect.

**Comment by gregw** [comment]
> I think the test failures are not a flake, the `SiweTest` was expecting the full URI in the redirect but now only gets the relative redirect.

Fixed.  Also those tests had bad ee10 packages.   Have you blogged about this feature yet?

**Review by janbartel — APPROVED** [review]

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java [hunk 1]
```diff
@@ -87,15 +87,15 @@ public class Server extends Handler.Wrapper implements Attributes
     private final MimeTypes.Mutable _mimeTypes = new MimeTypes.Mutable();
     private String _serverInfo = __serverInfo;
     private boolean _openEarly = true;
-    private boolean _stopAtShutdown;
+    private boolean _stopAtShutdown = false;
```
Comment (sbordet) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java]: Unnecessary.

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java [hunk 2]
```diff
@@ -87,15 +87,15 @@ public class Server extends Handler.Wrapper implements Attributes
...
     private boolean _dumpAfterStart;
     private boolean _dumpBeforeStop;
     private Handler _defaultHandler;
     private Request.Handler _errorHandler;
     private RequestLog _requestLog;
     private boolean _dryRun;
     private volatile DateField _dateField;
-    private long _stopTimeout;
+    private long _stopTimeout = 0;
```
Comment (sbordet) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java]: Unnecessary.

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DefaultHandler.java [hunk 3]
```diff
@@ -62,8 +62,8 @@ public class DefaultHandler extends Handler.Abstract
     private final long _faviconModifiedMs = (System.currentTimeMillis() / 1000) * 1000L;
     private final HttpField _faviconModified = new PreEncodedHttpField(HttpHeader.LAST_MODIFIED, DateGenerator.formatDate(_faviconModifiedMs));
     private ByteBuffer _favicon;
-    private boolean _serveFavIcon = true;
-    private boolean _showContexts = true;
+    private boolean _serveFavIcon;
+    private boolean _showContexts;
```
Comment (sbordet) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DefaultHandler.java:66]: The defaults in XML is `true`, so should be `true` here too (like it was before), to have the same behavior between standalone and embedded.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DefaultHandler.java (hunk 3 above)
Comment (gregw) [reply jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DefaultHandler.java:66]: It is either passed into the constructor or set to true by the default constructor, so it already defaults to true.

**Review by olamy — APPROVED** [review]

**Review by sbordet — APPROVED** [review]

**Review by lorban — APPROVED** [review]

**Pull request merged**